### PR TITLE
Spring 6 migration

### PR DESCRIPTION
**Pull request by fatroom**
## Description
This is to track the work on migration to spring 6.

## Motivation and Context
Ref: #1351 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-httpclient/src/main/java/org/zalando/riptide/httpclient/ApacheClientHttpRequestFactory.java [hunk 1]
```diff
@@ -2,16 +2,8 @@
...
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpHead;
-import org.apache.http.client.methods.HttpOptions;
-import org.apache.http.client.methods.HttpPatch;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.client.methods.HttpPut;
-import org.apache.http.client.methods.HttpTrace;
-import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.hc.client5.http.classic.HttpClient;
+import org.apache.hc.client5.http.classic.methods.*;
```
Comment (cberg-zalando) [inline riptide-httpclient/src/main/java/org/zalando/riptide/httpclient/ApacheClientHttpRequestFactory.java:6]: Not sure if you want to keep the project's standard of going without `*` imports.

File: riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/DefaultIdempotencyDetector.java [hunk 2]
```diff
@@ -1,30 +1,28 @@
 package org.zalando.riptide.idempotency;
 
 import org.apiguardian.api.API;
+import org.springframework.http.HttpMethod;
 import org.zalando.riptide.RequestArguments;
 
+import java.util.Set;
+
 import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.springframework.http.HttpMethod.*;
```
Comment (cberg-zalando) [inline riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/DefaultIdempotencyDetector.java:10]: Same as above regarding the `*` import.

File: riptide-idempotency/src/test/java/org/zalando/riptide/idempotency/IdempotencyPredicateTest.java [hunk 3]
```diff
@@ -4,36 +4,45 @@
...
 import org.zalando.riptide.RequestArguments;
 
 import java.util.function.Predicate;
+import java.util.stream.Stream;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.springframework.http.HttpMethod.GET;
-import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.HttpMethod.*;
```
Comment (cberg-zalando) [inline riptide-idempotency/src/test/java/org/zalando/riptide/idempotency/IdempotencyPredicateTest.java:17]: Here as well.

File: riptide-parent/pom.xml [hunk 4]
```diff
@@ -24,8 +24,8 @@
         <micrometer.version>1.10.3</micrometer.version>
         <mockito.version>4.11.0</mockito.version>
         <problem.version>0.27.1</problem.version>
-        <slf4j.version>1.7.36</slf4j.version>
-        <spring-boot.version>2.7.7</spring-boot.version>
+        <slf4j.version>2.0.6</slf4j.version>
+        <spring-boot.version>3.0.0</spring-boot.version>
```
Comment (cberg-zalando) [inline riptide-parent/pom.xml:28]: Should this be the latest Spring Boot 3.* version?

**Comment by fatroom** [comment]
@cberg-zalando this pr is dropped in favour to split work to less intrusive changes.

**Pull request closed without merging**